server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -152,20 +112,45 @@ func (w *Worker) work(ctx workflow.Context, revision string) error {
 
 	logger.Info(ctx, fmt.Sprintf("latest deployed revision %s", deployedRevision))
 
-	// TODO: fill in the rest
-	// Spin up a child workflow to handle Terraform operations
-	//childWorkflowOptions := workflow.ChildWorkflowOptions{
-	//	WorkflowID: id.String(),
-	//}
-	//ctx = workflow.WithChildOptions(ctx, childWorkflowOptions)
-	//terraformWorkflowRequest := terraform.Request{
-	//	Repo: w.Repo,
-	//	Root: w.Root,
-	//}
-	//err = workflow.ExecuteChildWorkflow(ctx, w.TerraformWorkflow, terraformWorkflowRequest).Get(ctx, nil)
-	//if err != nil {
-	//	return errors.Wrap(err, "executing child terraform workflow")
-	//}
+	ctx = workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{
+		WorkflowID: id.String(),
+	})
+	terraformWorkflowRequest := terraform.Request{
+		Repo: w.Repo,
+		Root: w.Root,
+	}
+
+	future := workflow.ExecuteChildWorkflow(ctx, w.TerraformWorkflow, terraformWorkflowRequest)
+	return w.awaitTerraformWorkflow(ctx, future, checkRunID)
+}
+
+func (w *Worker) awaitTerraformWorkflow(ctx workflow.Context, future workflow.ChildWorkflowFuture, checkRunID int64) error {
+	var childWE workflow.Execution
+	future.GetChildWorkflowExecution().Get(ctx, &childWE)
+	ch := workflow.GetSignalChannel(ctx, state.WorkflowStateChangeSignal)
```

Comment:
I don't see where signals are being sent, is that in a followup PR/can you explain how they will be sent?

Reply:
Yeah they'll be in a followup to limit the scope of this PR, I can leave a comment with a brief one liner.